src/adapters/github.js

```diff
@@ -149,10 +153,93 @@ class GitHub extends PjaxAdapter {
-      else cb(null, res.tree)
+      else {
+        if (!opts.repo.pull) cb(null, res.tree)
+        else {
+          this._getPatch(opts, (patchErr, patchRes) => {
+            const diffExists = patchRes && Object.keys(patchRes).length > 0
+            if (patchErr || !diffExists) cb(null, res.tree)
+            else {
+              // Filter tree to only include files and directories that are included in the patch
+              const filteredTree = res.tree
+                  .filter((node) => {
+                    return patchRes[node.path] !== undefined
+                  })
+                  .map((node) => {
+                    const patch = patchRes[node.path]
+                    node.patch = patch
+                    return node
+                  })
+              cb(null, filteredTree)
+            }
+          })
+        }
+      }
     })
   }
 
+  /**
+   * Get files that were patched in Pull Request.
+   * The diff map that is returned contains changed files, as well as the parents of the changed files.
+   * This allows the tree to be filtered for only folders that contain files with diffs.
+   * @param {Object} opts: {
+   *                  path: the starting path to load the tree,
+   *                  repo: the current repository,
+   *                  node (optional): the selected node (null to load entire tree),
+   *                  token (optional): the personal access token
+   *                 }
+   * @param {Function} cb(err: error, diffMap: Object)
+   */
+   _getPatch(opts, cb) {
+    const {pull} = opts.repo
+    this._get(`/pulls/${pull}/files`, opts, (err, res) => {
+      if (err) cb(err)
+      else {
+        const diffMap = {}
+        res.forEach(file => {
+          // Grab parent folder path
+          const folderPath = file.filename.split('/').slice(0,-1).join('/')
+          // Record file patch info
+          diffMap[file.filename] = {
+            action: file.status,
+            additions: file.additions,
+            deletions: file.deletions,
+            filename: file.filename,
+            path: file.path,
+            sha: file.sha,
+          }
+          // Record ancestor folder patch info
+          const split = folderPath.split('/')
+          // Start at root folder and construct path
+          split.reduce((path, curr) => {
+            if (path.length) {
+              path = `${path}/${curr}`
+            }
+            else {
+              path = `${curr}`
+            }
+            // Path already has been recorded, accumulate changes
+            if (typeof diffMap[path] === 'object') {
+              diffMap[path].additions += file.additions
+              diffMap[path].deletions += file.deletions
+              diffMap[path].changes++
+            }
+            // Path is new
+            else {
+              diffMap[path] = {
+                additions: file.additions,
+                deletions: file.deletions,
+                changes: 1,
```

Comment:
Related to my previous comment, consider changing this property to `files` or similar.